launcher/modManager/cmodlistview_moc.cpp

```diff
@@ -617,6 +607,58 @@ void CModListView::on_installButton_clicked()
 	}
 }
 
+void CModListView::on_installFromFileButton_clicked()
+{
+	QString filter = tr("All supported files") + " (*.h3m *.vmap *.h3c *.vcmp *.zip *.json);;" + tr("Maps") + " (*.h3m *.vmap);;" + tr("Campaigns") + " (*.h3c *.vcmp);;" + tr("Configs") + " (*.json);;" + tr("Mods") + " (*.zip)";
+	QStringList files = QFileDialog::getOpenFileNames(this, tr("Select files (configs, mods, maps, campaigns) to install..."), QDir::homePath(), filter);
+
+	for (const auto & file : files)
+	{
+		QUrl url = QUrl::fromLocalFile(file);
+		manualInstallFile(url);
+	}
+}
+
+void CModListView::manualInstallFile(QUrl url)
+{
+	QString urlStr = url.toString();
+	QString fileName = url.fileName();
+	if(urlStr.endsWith(".zip", Qt::CaseInsensitive))
+		downloadFile(fileName.toLower()
+			// mod name currently comes from zip file -> remove suffixes from github zip download
+			.replace(QRegularExpression("-[0-9a-f]{40}"), "")
+			.replace(QRegularExpression("-vcmi-.+\\.zip"), ".zip")
+			.replace("-main.zip", ".zip")
+			, urlStr, "mods", 0);
+	if(urlStr.endsWith(".json", Qt::CaseInsensitive))
+	{
+		QDir configDir(QString::fromStdString(VCMIDirs::get().userConfigPath().string()));
+		QStringList configFile = configDir.entryList({fileName}, QDir::Filter::Files);
+		if(!configFile.empty())
+		{
+			if(QMessageBox::warning(this, tr("Replace config file?"), tr("Do you want to replace ") + configFile[0] + "?", QMessageBox::Yes | QMessageBox::No, QMessageBox::No) == QMessageBox::Yes)
```

Comment:
```suggestion
			if(QMessageBox::warning(this, tr("Replace config file?"), tr("Do you want to replace %1?").arg(configFile[0]), QMessageBox::Yes | QMessageBox::No, QMessageBox::No) == QMessageBox::Yes)
```

and I think it's better to extract dialog result to a variable as the condition is quite long

Reply:
> and I think it's better to extract dialog result to a variable as the condition is quite long

whenever you decide to ignore comments, please state the reason explicitly

Reply:
Done